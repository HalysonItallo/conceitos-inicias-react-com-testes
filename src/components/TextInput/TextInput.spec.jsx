import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { it, vi } from 'vitest'
import { TextInput } from '.'

describe('<TextInput />', () => {
  it('should have a value of searchValue', () => {
    const fn = vi.fn()

    render(
      <TextInput
        onChange={fn}
        value={'test'}
        placeholder={'type your search'}
      />
    )

    const input = screen.getByPlaceholderText(/type your search/i)

    expect(input.value).toBe('test')
  })

  it('should call handleChange function on each key pressed', async () => {
    const fn = vi.fn()

    render(<TextInput onChange={fn} placeholder={'type your search'} />)

    const input = screen.getByPlaceholderText(/type your search/i)

    const value = 'o valor'

    await userEvent.type(input, value)

    expect(input).toHaveValue(value)
    expect(fn).toHaveBeenCalledTimes(value.length)
  })

  it('should match snapshot', () => {
    const fn = vi.fn()
    const { container } = render(
      <TextInput onChange={fn} placeholder={'type your search'} />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
