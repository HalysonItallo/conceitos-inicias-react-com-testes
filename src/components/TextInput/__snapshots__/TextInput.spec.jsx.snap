// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<TextInput /> > should match snapshot 1`] = `
<input
  class="text-input"
  placeholder="type your search"
  type="search"
  value=""
/>
`;
